lisa/tools/python.py

```diff
@@ -49,17 +49,40 @@ def _install(self) -> bool:
         self.node.os.install_packages(package_name)
         return self._check_exists()
 
-    def install_packages(self, packages_name: str) -> None:
-        cmd_result = self.run(
-            f"install -q {packages_name}",
-        )
+    def install_packages(
+        self,
+        packages_name: str,
+        install_path=""
+    ) -> None:
+        node = self.node
+        cmd_line = f"install -q {packages_name}"
+
+        if install_path != "":
+            tagert_path = install_path + "/python_packages"
+            node.tools[Mkdir].create_directory(tagert_path, sudo=True)
+            tmp_path = install_path + "/tmp"
+            node.tools[Mkdir].create_directory(tmp_path, sudo=True)
+            node.execute(
+                f"chmod -R 777 {tagert_path}; chmod -R 777 {tmp_path}",
+                shell=True,
+                sudo=True,
+            )
+            cmd_line = cmd_line + f" --target {tagert_path} --cache-dir={tmp_path} -b {tmp_path}"
+            env_list = {"TMPDIR": tmp_path}
+        else:
+            env_list = {}
```

Comment:
1. The variable name should be `envs`. Avoid to include type in name, an it's not a "list".
2. The envs can be initialized before the `if` block. So the `else` block is not necessary. The logic will be simpler.